url2bc31e3489_ramonli_RabbitMQResearch_tgz-pJ8-com_mpos_lottery_te_thirdpartyservice_amqp_protobuf_CancellationMessagePulisherJ8/src/com/mpos/lottery/te/thirdpartyservice/amqp/RabbitFirehoseMessageConsumerMain.java

```diff
@@ -0,0 +1,30 @@
+package com.mpos.lottery.te.thirdpartyservice.amqp;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+public class RabbitFirehoseMessageConsumerMain {
+    private static Log logger = LogFactory.getLog(RabbitFirehoseMessageConsumerMain.class);
+    public static final String EXCHANGE_NAME = "amq.rabbitmq.trace";
+
+    public static void main(String[] argv) throws Exception {
+        RabbitMessageConsumer main = new RabbitMessageConsumer("192.168.2.152", true,
+                "amqp", "amqp");
+        main.consume(EXCHANGE_NAME, RabbitMessagePublisher.class.getName(), new String[] { "publish.*",
+                "deliver.*" }, 1);
+    }
+
+    public static void deleteQueue() throws Exception {
+        ConnectionFactory factory = new ConnectionFactory();
+        factory.setHost("192.168.2.152");
+        Connection connection = factory.newConnection();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *connection*. The resource is referenced at lines 25, 26, 27. The resource is closed at line 28. There are other execution paths that do not contain closure statements, for example, when Channel.queueDelete throws an exception. Ensure *connection* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).